DEVX-1130: Add C3 to Multi Region Replication demo (#886)
confluentinc/examples: multiregion/docs/multiregion.rst

You can choose how to replicate data, synchronously or asynchronously, on a per |ak| topic basis.
It provides good durability guarantees and makes disaster recovery (DR) much easier.

> We will use a dedicated metrics cluster running |c3| to monitor the Multi-Region Cluster.

Reviewer: ```suggestion
This tutorial uses a dedicated |ak| cluster backing |c3|, which monitors the multi-region cluster.
```

         Topic: multi-region-default PartitionCount: 1   ReplicationFactor: 4    Configs: min.insync.replicas=1,confluent.placement.constraints={"version":1,"replicas":[{"count":2,"constraints":{"rack":"west"}}],"observers":[{"count":2,"constraints":{"rack":"east"}}]}
            Topic: multi-region-default Partition: 0    Leader: 2   Replicas: 2,1,3,4   Isr: 2,1    Offline:    Observers: 3,4

#. View the topic replica placement in Confluent Control Center:

   Navigate to the C3 UI at http://localhost:9021.

   |C3 cluster navigation|

>    Click on the "mrc" cluster, then make your way to the "Topics" section.

Reviewer: It would be helpful to explicitly explain the difference between the 2 clusters

Author: the latest docs preview link reflects an explanation of the 2 clusters

Reviewer: Awesome, thanks!

         Topic: multi-region-default PartitionCount: 1   ReplicationFactor: 4    Configs: min.insync.replicas=1,confluent.placement.constraints={"version":1,"replicas":[{"count":2,"constraints":{"rack":"west"}}],"observers":[{"count":2,"constraints":{"rack":"east"}}]}
            Topic: multi-region-default Partition: 0    Leader: 2   Replicas: 2,1,3,4   Isr: 2,1    Offline:    Observers: 3,4

#. View the topic replica placement in |c3|:

   Navigate to the C3 UI at http://localhost:9021.

   |C3 cluster navigation|

>    You will notice 2 clusters, "mrc" which is the multiregion cluster and "metrics" which is a dedicated metrics cluster that runs Control Center. By hosting Control Center on its own Kafka cluster, it is independent of the availability of the production cluster it is monitoring. We will be working on topics within the "mrc" cluster for the remainder of this tutorial. Click on the "mrc" cluster, then make your way to the "Topics" section.

Reviewer: ```suggestion
   Notice two clusters: "mrc" which is the multiregion cluster, and "metrics" which is a dedicated metrics cluster that runs |c3|. By backing |c3| to its own |ak| cluster, it has no dependency on the availability of the production cluster it is monitoring. The remainder of this tutorial works on topics within the "mrc" cluster. Click on the "mrc" cluster, then make your way to the "Topics" section.
```

   |C3 topics overview|

>    Click on each topic to see details about the replica and observer placement. This will mirror the CLI output above. Below is an example of the multi-region-async topic.

Reviewer: ```suggestion
   Click on each topic to see details about the replica and observer placement. |c3| matches the CLI output above. Below is an example of the ``multi-region-async`` topic.
```

   |multi-region-async replicas|

#. Observe the following:

   - The ``multi-region-async``, ``multi-region-async-op-under-min-isr``, ``multi-region-async-op-under-replicated``, ``multi-region-async-op-leader-is-observer`` and ``multi-region-default`` topics have replicas
     across ``west`` and ``east`` regions, but only 1 and 2 are in the ISR, and 3 and

>      4 are observers. This can be observed via the CLI output or C3.

Reviewer: ```suggestion
     4 are observers. This can be observed via the CLI output or |c3|.
```

      multi-region-async-op-leader-is-observer: 0
      multi-region-default: 0

> #. Metrics relating to partitions, followers and observers are viewable in the "Topics" section of the C3 UI.

Reviewer: C3 doesn't show exactly the same JMX metrics as the output above.  Is the screenshot needed at all if this is the same diagram as shown earlier?

Author: It has been removed and the latest docs preview link reflects that.

      Topic: multi-region-default	PartitionCount: 1	ReplicationFactor: 4	Configs: min.insync.replicas=1,confluent.placement.constraints={"version":1,"replicas":[{"count":2,"constraints":{"rack":"west"}}],"observers":[{"count":2,"constraints":{"rack":"east"}}]}
      	Topic: multi-region-default	Partition: 0	Leader: 2	Replicas: 1,2,3,4	Isr: 2	Offline: 1	Observers: 3,4

> #. Verify similar replica placement in C3. Note that it may take up to 5 minutes for C3 to properly report the new topic stats.

Reviewer: ```suggestion
#. Verify similar replica placement in |c3|. Note that it may take up to 5 minutes it to properly report the new topic stats.
```

Reviewer: @awalther28 note that `C3` is an internal term, but we should use `Control Center` or `Confluent Control Center` in documentation.  

```
.. |c3-short| replace:: Control Center
.. |c3| replace:: Confluent Control Center
```

      Topic: multi-region-default PartitionCount: 1   ReplicationFactor: 4    Configs: min.insync.replicas=1,confluent.placement.constraints={"version":1,"replicas":[{"count":2,"constraints":{"rack":"west"}}],"observers":[{"count":2,"constraints":{"rack":"east"}}]}
         Topic: multi-region-default Partition: 0    Leader: none    Replicas: 2,1,3,4   Isr: 1  Offline: 2,1    Observers: 3,4

> #. Given five minutes, you should see something similar to below in C3. The numbers may be off while the cluster stabilizes.

Reviewer: ```suggestion
#. After the |c3| cluster metrics stabilize in about five minutes, you should see output similar to below.
```

     region is restored. Leader election is not required for this demotion
     process, it will happen as soon as the failed region is restored.

>    - The C3 topics page is the same as it was at the start of this tutorial.

Reviewer: ```suggestion
   - The |c3| topics page is the same as it was at the start of this tutorial.
```

.. |Multi-region Architecture|
image:: images/multi-region-base-v2.png
   :alt: Multi-region Architecture

.. |Follower_Fetching|
image:: images/Follower_Fetching.png
   :alt: Follower fetching

.. |Multi-region latencies|
image:: images/multi-region-latencies-v2.png
   :alt: Multi-region latencies

.. |Multi-region topic replicas|
image:: images/multi-region-topic-replicas-v2.png
   :alt: Multi-region topic replicas

.. |C3 cluster navigation|
image:: images/c3-cluster-navigation.png
   :alt: C3 cluster navigation

.. |C3 topics overview|
image:: images/c3-topics-overview.png
   :alt: C3 topics overview

.. |multi-region-async replicas|
image:: images/multi-region-async-placement.png

>    :alt: multi-region-async replicas

Reviewer: @awalther28 WDYT about adding the following to make the graphic smaller?

```
      :width: 500px
```

Author: The smaller picture looks much better. 
<img width="1108" alt="Screen Shot 2021-01-04 at 11 56 39 AM" src="https://user-images.githubusercontent.com/12937975/103569093-eedbda00-4e83-11eb-84d4-ad78f22732f7.png">
